server/endpoints/Search.ts

```diff
@@ -223,7 +223,8 @@ export const getCoursesByProfessor: Endpoint<Search> = {
       let courses = [];
       const regex = new RegExp(/^(?=.*[A-Z0-9])/i);
       if (regex.test(search.query)) {
-        courses = await Classes.find({ classProfessors: search.query }).exec();
+        const professorRegex = search.query.replace('+', '.*.');
+        courses = await Classes.find({ classProfessors: { $regex: professorRegex, $options: "i" } }).exec();
```

Comment:
what is this for?

Reply:
I don't know, I did not implement this. Looks to be for finding the professors for a course?

Reply:
might have to rebase